src/docs/arch-btree.dox

```diff
@@ -0,0 +1,87 @@
+/*! @arch_page arch-btree B-Trees
+
+WiredTiger represents database tables using a B-Tree data structure (\c WT_BTREE in
+\c btree.h), which is made up of nodes that are page structures. The root and internal
+pages only store keys and references to other pages, while leaf pages store keys and
+values. Pages are populated with records as users insert data into the database. Records
+are maintained in sorted order according to their key, and pages will split once their
+configured limit is reached, causing the B-Tree to expand. Pages have an in-memory
+representation as well as an on-disk representation. The focus here will be on
+the in-memory representation of the B-Tree and its pages as defined in \c btmem.h. The
+on-disk representation is discussed in the @ref arch-data-file page.
+
+@section btree_btree_data_source B-Tree Data Source (WT_BTREE)
+
+As discussed in the @ref arch-dhandle page, data handles (dhandles), are generic
+containers used to access various types of data sources. Dhandles that represent
+B-Trees contain a pointer to the \c WT_BTREE. At a high-level, the \c WT_BTREE
+contains a memory cache of key value pairs, along with functions to read and
+write data as needed to and from the data file. It also contains the specific WT_BTREE
+type, a reference to the root page on disk for access to the underlying data, and
+information used to maintain the B-Tree structure. The different B-Tree types support
+different access methods; row-store is most commonly used (\c WT_BTREE_ROW), and
+column-store is available with either fixed-length records (\c WT_BTREE_COL_FIX) or
+variable-length records (\c WT_BTREE_COL_VAR) (see @ref arch-row-column for more details).
+
+@section btree_btree_in_memory_representation B-Tree In-Memory Representation
+
+B-Trees can grow to a very large size, and the space in memory is
+generally not large enough to hold all the pages of the B-Tree. To access a page in
+the B-Tree, we require a \c WT_REF which tracks whether the page has or has not been
+loaded from storage. Once the page is loaded, the \c WT_REF will have a valid \c WT_PAGE
+pointer which represents the in-memory page. The \c WT_BTREE structure contains a
+\c WT_REF or reference structure that points to the root page of the given tree. Other
```

Comment:
Move the "or reference structure" to after the first use of `WT_REF` in this paragraph.